reference/features/search_parameters.md

```diff
@@ -2,132 +2,163 @@
+| **[cropLength](/reference/features/search_parameters.md#crop-length)**                        | Length used to crop field values                                                                | `200`         |
+| **[attributesToHighlight](/reference/features/search_parameters.md#attributes-to-highlight)** | Attributes whose values will contain highlighted matching terms                                 | `null`        |
+| **[matches](/reference/features/search_parameters.md#matches)**                               | Defines whether an object that contains information about the matches should be returned or not | `false`       |
+
+::: warning
+All parameters must be **URL-encoded** when searching an index with a `GET` route.
+
+This is not necessary when using the `POST` route or one of our SDKs.
+:::
+
+## Examples
+
+Results can be paginated using the `limit` and `offset` query parameters.
+
+<CodeSamples id="search_guide_1" />
+
+You can filter results using the `filters` query parameter.
+
+<CodeSamples id="search_guide_2" />
 
 ## Query (q)
 
-This is the string used by the search engine to find relevant documents. Search queries are [tokenized](/reference/under_the_hood/tokenization.md).
+**Parameter**: `q`
+**Expected value**: any string
+**Default value**: `null`
+
+Sets the queried terms.
 
 ::: warning
-MeiliSearch only considers the first ten words of any given search query. This is necessary in order to deliver a [fast type-as-you-search experience](/learn/what_is_meilisearch/philosophy.md#front-facing-search) in a consistent way.
+MeiliSearch only considers the first ten words of any given search query. This is necessary in order to deliver a [fast search-as-you-type experience](/learn/what_is_meilisearch/philosophy.md#front-facing-search).
 :::
 
-`q=<String>`
+### Example
 
-- `<String>` (String)
+You can search for films mentioning `shifu` by setting the `q` parameter:
 
-  The query string.
-
-#### Example
+<CodeSamples id="search_parameter_guide_query_1" />
 
-Suppose you would like to search `shifu` in a movie database. You would send the following:
+This will give you a list of documents that contain that string in at least one of its attributes.
 
-<CodeSamples id="search_parameter_guide_query_1" />
+```json
+{
+  "hits": [
+    {
+      "id":"50393",
+      "title":"Kung Fu Panda Holiday",
+      "poster":"https://image.tmdb.org/t/p/w500/rV77WxY35LuYLOuQvBeD1nyWMuI.jpg",
+      "overview":"The Winter Feast is Po's favorite holiday. Every year he and his father hang decorations, cook together, and serve noodle soup to the villagers. But this year Shifu informs Po that as Dragon Warrior, it is his duty to host the formal Winter Feast at the Jade Palace. Po is caught between his obligations as the Dragon Warrior and his family traditions: between Shifu and Mr. Ping.",
+      "release_date":1290729600,
+      "genres":["Animation","Family","TV Movie"]
+    },
+    …
+  ],
+  …
+  "query":"shifu"
+}
+```
```

Comment:
This example is very long, I regret using that one. If you want to use a smaller one I think it would be more clear on first sight

Reply:
[We are discussing decoupling examples from the `movies.json` dataset,](https://github.com/meilisearch/documentation/issues/1026) so that anything outside e.g. the Quick Start tutorial refers to a non-existent dataset. This would allow us to have very focussed examples; the above bit, for instance, could be abbreviated to:

```json
    {
      "id":"50393",
      "title":"Kung Fu Panda Holiday",
      "overview":"  <em>Shifu</em> informs Po that as Dragon Warrior, it is his duty to host the formal Winter Feast at the Jade Palace.",
      "genres":["Animation","Family","TV Movie"]
    }
```